x-pack/plugins/alerting/server/lib/snooze/is_snooze_active.ts

```diff
@@ -54,8 +57,27 @@ export function isSnoozeActive(snooze: RuleSnoozeSchedule) {
   return null;
 }
 
-export function parseByWeekday(byweekday: Array<string | number>): ByWeekday[] {
+export function parseByWeekday(
+  byweekday: Array<string | number>,
+  dtstart: string,
+  tzid: string
+): ByWeekday[] {
   const rRuleString = `RRULE:BYDAY=${byweekday.join(',')}`;
   const parsedRRule = rrulestr(rRuleString);
-  return parsedRRule.origOptions.byweekday as ByWeekday[];
+  const newbyweekday = parsedRRule.origOptions.byweekday as Weekday[];
+  const startMomentDay = moment(dtstart).tz(tzid).isoWeekday();
+  const startMomentUTCDay = moment(dtstart).tz('UTC').isoWeekday();
+  if (startMomentDay !== startMomentUTCDay) {
+    return newbyweekday.map((d) => shiftWeekday(d, startMomentDay - startMomentUTCDay));
+  }
+  return newbyweekday;
+}
+
+function shiftWeekday(d: Weekday, diff: number) {
```

Comment:
Im wondering if we can write a test for this, I think it might be tricky to do it topdown since we can't mock timezones, but maybe a simple test with the diff would be nice.